master/internal/stream/test_util.go

```diff
@@ -12,10 +13,47 @@ import (
 
 const (
 	channelBufferSize = 10
+	syncKey           = "sync_msg"
 )
 
-// simpleUpsert is for testing and just returns the preparable message that the streamer sends.
-func simpleUpsert(i stream.PreparableMessage) interface{} {
+// testPrepareFunc returns a string representation of known messages;
+// otherwise, returns the preparable message that the streamer sends.
+func testPrepareFunc(i stream.PreparableMessage) interface{} {
+	switch msg := i.(type) {
+	case stream.UpsertMsg:
+		switch typedMsg := msg.Msg.(type) {
+		case *TrialMsg:
+			return fmt.Sprintf(
+				"%s (%d): %s %d %d",
```

Comment:
as I think about this more, I'm torn over this format because it makes the testing code a lot simpler but less clear. we won't know exactly what broke a failed test until we find our way to this function. What do you think about making it formatted like: 
"key: %s, id: %s, state: %s", like a stringified map?

Reply:
Sounds good to me!